client/src/pages/MySitters/MySitters.tsx

```diff
@@ -1,33 +1,182 @@
+import BookingsDatePicker from "../../components/BookingsDatePicker/BookingsDatePicker";
+
+const mockImg = 'https://badairies.co.uk/assets/admin/plugins/images/users/4.jpg';
 
 export default function MySitters(): JSX.Element {
   const classes = useStyles();
 
   const [requests, setRequests] = useState<RequestData[]>([]);
+  const [dates, setDates] = useState<Date[]>([])
+  const [cardAction, setCardAction] = useState<boolean>(false);
+  const [selectedBooking, setSelectedBooking] = useState<string>("");
 
   useEffect(() => {
     getRequests()
-      .then(data => setRequests(data));
+      .then(data => {
+        setRequests(data)
+        const dates: Date[] = [];
+        data.map(booking => {
+          if (booking.accepted) {
+            const bookingDate = new Date(booking.start)
+            dates.push(bookingDate)
+          }
+        });
+        setDates(dates);
+    });
   }, []);
+  
+  const currentDate = new Date();
+
+  const nextBookingFinder = (bookings: RequestData[]) => {
+    const nextBooking: RequestData[] = []
+    bookings.forEach((booking) => {
+      if (booking.accepted && !nextBooking.length) {
+        nextBooking.push(booking)
+      }
+    });
+    return nextBooking;
+  };
+
+  const currentBookings = requests.filter((request) => {
+    const requestDate = new Date(request.start);
+    return requestDate >= currentDate;
+  });  
+  
+  const nextBooking: RequestData[] = nextBookingFinder(currentBookings);
+
+  const pastBookings = requests.filter((request) => {
+    const requestDate = new Date(request.start);
+    return (requestDate < currentDate && request.accepted);
+  })
+
+  // function to display bookings dates
+  const createDate = (dateStart: any, dateEnd: any) => {
+    dateStart = new Date(dateStart);
+    dateEnd = new Date(dateEnd);
+    const day = dateStart.getDate();
+    let month = dateStart.toLocaleDateString("us-US", {month: "long"});
+    month = month.charAt(0).toUpperCase() + month.slice(1);
+    const year = dateStart.getFullYear();
+    const timeStart = dateStart.getHours();
+    const timeEnd = dateEnd.getHours();
+    return `${day}, ${month} ${year}, ${timeStart}-${timeEnd} HS.`
+  }
+
+  // display card action
+  const displayCardActions = (booking: RequestData): void => {
+    if (!booking.accepted && !booking.declined) {
+      setCardAction(true);
+      setSelectedBooking(booking._id)
+    }
+  };
 
-  console.log(requests)
+  // update selected request
+  const updateSelectedRequest = (bookingId: string, state: string) => {
+    updateRequest(bookingId, state)
+    .then(() => setCardAction(false))
+    .then(() => getRequests()
+    .then(data => {
+      setRequests(data)
+      const dates: Date[] = [];
+      data.map(booking => {
+        if (booking.accepted) {
+          const bookingDate = new Date(booking.start)
+          dates.push(bookingDate)
+        }
+      });
+      setDates(dates);
+    })
+    )
```

Comment:
We should also use async/await here, this is more consistent with the overall codebase. We want to strive for increasing codebase quality and not mixed which is decreasing.

Reply:
Done